Lib/asyncio/sslproto.py

```diff
@@ -569,9 +570,26 @@ def _start_handshake(self):
         self._write_backlog.append((b'', 1))
         self._loop.call_soon(self._process_write_backlog)
 
+        # in case if waiter get cancelled before handshake completes
+        if self._waiter is not None and not self._waiter.done():
+            self._waiter.add_done_callback(self._handshake_waiter)
+
+    def _handshake_waiter(self, waiter):
+        if self._in_handshake and self._transport is not None:
+            if waiter.cancelled():
+                exc = base_futures.CancelledError()
+            else:
+                exc = waiter.exception()
+
+            if exc is not None:
+                self._on_handshake_complete(exc)
```

Comment:
Can the `waiter` here be in a resolved state without an exception set?  I.e. is `else` clause possible here?

Reply:
2020 and I'm still seeing this issue when the site returns an invalid certificate. The handshake is aborted and the socket isn't released. I'm on Pyhton 3.6.8 with the latest asyncio and aiohttp.